OpenCL.pc.in

```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
```

Comment:
Previously, this used `CMAKE_INSTALL_FULL_*`, now it joins together `CMAKE_INSTALL_PREFIX` and `CMAKE_INSTALL_*`.

Actually, though, *both* approaches are kinda broken. Meson has a robust pkg-config generator that handles these edge cases and a lot more for you, but unfortunately CMake does not... however, you can work around this with a bit of scripting.

Please see https://github.com/jtojnar/cmake-snips/#assuming-cmake_install_dir-is-relative-path and including the very permissively licensed `JoinPaths.cmake` module, and using it before `configure_file()`.

Reply:
Thanks for the input. I was worrying about something like that, and it was the reason I tried going with `CMAKE_INSTALL_FULL_*` first. I will defer to our CMake expert (:wink: @MathiasMagnus) for how to integrate the above.

For reference, here is an example of a 'correct' integration:
https://github.com/fmtlib/fmt/pull/1657/files

Reply:
Imho, the existing changes are "good enough" for the vast majority of use cases (that install headers and libraries at the same prefix).
Would it be ok to use the existing changes for now and fix the remaining use case (where headers or libraries are installed outside the prefix location) at a later time?

Reply:
On Windows you actually want to pretend that prefix is `${pcfiledir}\..\..` for as many times `\..` as there are relative links, but that's rapidly approaching and likely beyond the feasibility of what you can do without python or C++ ergonomics. If you want proper pkg-config support either use Meson or wait 20 years for cmake to decide they are willing to implement pkg-config generation? :shrug:

But actually I think windows should work fine with that module too, it allows `/` paths and cmake itself handles IS_ABSOLUTE (correctly, I expect).

So, the cmake snip isn't any worse on Windows than not using it.

...

Is this PR really being held up by my comment? I really, *really* doubt it.

Reply:
@Kerilk

> I will defer to our CMake expert (😉 @MathiasMagnus) for how to integrate the above.
> 
> For reference, here is an example of a 'correct' integration:
> https://github.com/fmtlib/fmt/pull/1657/files

Since you already know and give examples of integrating it, I don't see why you need a "cmake expert" to verify that it's okay to do.

Is cmake such a complicated build system that it's too dangerous to change anything whatsoever unless someone with a degree in cmake agrees with not only the what, but also the where and the how and the why?

I think it's safe enough to do it the obvious way :) and really, the best way to get a review from a cmake expert is to do an initial implementation and then ask for a ***review***. :)

Seems like the PR is currently just waiting on a known and expected change to be trivially implemented and pushed, so it would be great to get this moving forward again in the simple and obvious way. :)